File: src/nodes/if_else.zig
const nd = @import("node.zig");

pub const NodeIfElse = struct {
    condition: *nd.Node,
    if_body: *nd.Node,
    else_body: *nd.Node,

    pub fn init(condition: *nd.Node, if_body: *nd.Node, else_body: *nd.Node) NodeIfElse {
        return NodeIfElse{ .condition = condition, .if_body = if_body, .else_body = else_body };
    }

    pub fn get_condition(self: *const NodeIfElse) *nd.Node {
        return self.condition;
    }

    pub fn get_if_body(self: *const NodeIfElse) *nd.Node {
        return self.if_body;
    }

    pub fn get_else_body(self: *const NodeIfElse) *nd.Node {
        return self.else_body;
    }
};

File: src/nodes/if_else.rs
use crate::nodes::node::Node;

pub struct NodeIfElse {
    condition: Node,
    if_body: Node,
    else_body: Node,
}

impl NodeIfElse {
    pub fn init(condition: Node, if_body: Node, else_body: Node) -> NodeIfElse {
        NodeIfElse { condition, if_body, else_body }
    }
    
    pub fn get_condition(&self) -> &Node {
        &self.condition
    }
    
    pub fn get_if_body(&self) -> &Node {
        &self.if_body
    }
    
    pub fn get_else_body(&self) -> &Node {
        &self.else_body
    }
}